chalk-rust-ir/src/lib.rs

```diff
@@ -97,8 +125,20 @@ pub struct TraitDatum<I: Interner> {
     /// chalk we add annotations like `#[auto]`.
     pub flags: TraitFlags,
 
-    /// The id of each associated type defined in the trait.
+    /// The ids for the associated type members of the trait. The details of
+    /// each type can be found by invoking the [`associated_ty_data`] method.
+    ///
+    /// [`associated_ty_data`]: ../chalk_solve/trait.RustIrDatabase.html#tymethod.associated_ty_data
     pub associated_ty_ids: Vec<AssocTypeId<I>>,
+
+    /// A list of the traits that are "well known" to chalk, which means that
+    /// the chalk-solve crate has special, hard-coded impls for them.
```

Comment:
I'm wondering if it would be better to put these comments on the definitions of the members' types rather than on the members themselves

Reply:
This comment seems better for the `WellKnownTrait` type. I imagine the comment for the member would be:

`None` if this is a normal, user-defined trait; `Some` if this is a well-known trait.